File: container-documentapi/src/vespa/documentapi/messagebus/messages/queryresultmessage.cpp
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "queryresultmessage.h"

namespace documentapi {

QueryResultMessage::QueryResultMessage() = default;

QueryResultMessage::QueryResultMessage(const vdslib::SearchResult & result, const vdslib::DocumentSummary & summary) :
    VisitorMessage(),
    _searchResult(result),
    _summary(summary)
{}

QueryResultMessage::~QueryResultMessage() = default;

DocumentReply::UP
QueryResultMessage::doCreateReply() const
{
    return DocumentReply::UP(new VisitorReply(DocumentProtocol::REPLY_QUERYRESULT));
}

uint32_t
QueryResultMessage::getApproxSize() const
{
    return getSearchResult().getSerializedSize() + getDocumentSummary().getSerializedSize();
}

uint32_t
QueryResultMessage::getType() const
{
    return DocumentProtocol::MESSAGE_QUERYRESULT;
}

}

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/QueryResultMessage.java
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.documentapi.messagebus.protocol;

import com.yahoo.vdslib.SearchResult;
import com.yahoo.vdslib.DocumentSummary;

/**
 */
public class QueryResultMessage extends VisitorMessage {

    private SearchResult searchResult = null;
    private DocumentSummary summary = null;

    public SearchResult getResult() {
        return searchResult;
    }

    public DocumentSummary getSummary() {
        return summary;
    }

    public void setSearchResult(SearchResult result) {
        searchResult = result;
    }

    public void setSummary(DocumentSummary summary) {
        this.summary = summary;
    }

    @Override
    public DocumentReply createReply() {
        return new VisitorReply(DocumentProtocol.REPLY_QUERYRESULT);
    }

    @Override
    public int getType() {
        return DocumentProtocol.MESSAGE_QUERYRESULT;
    }
}
